### PR TITLE
Made udp_server.c compile for iOS

### PR DESCRIPTION
**Pull request by murgatroid99**
This is a temporary workaround for #3041.

### CODE REVIEW

File: src/core/iomgr/udp_server.c [hunk 1]
```diff
@@ -235,8 +235,10 @@ static int prepare_socket(int fd, const struct sockaddr *addr, int addr_len) {
   rc = setsockopt(fd, IPPROTO_IP, IP_PKTINFO, &get_local_ip,
                   sizeof(get_local_ip));
   if (rc == 0 && addr->sa_family == AF_INET6) {
+#if !TARGET_OS_IPHONE
```
Comment (ctiller) [inline src/core/iomgr/udp_server.c]: ifndef TARGET_OS_IPHONE?

Comment (murgatroid99) [inline src/core/iomgr/udp_server.c]: port_platform.h checks it with `#if TARGET_OS_IPHONE`. I just negated that.

Comment (ctiller) [inline src/core/iomgr/udp_server.c]: Then you'd probably need at least a defined(__APPLE__) also, right?

Comment (murgatroid99) [inline src/core/iomgr/udp_server.c]: Undefined variables default to 0.

**Comment by ctiller** [comment]
LGTM then. Waiting on Jenkins.

**Pull request merged**